src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -878,6 +910,9 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             //      becomes a no-op.
             this->Focus(FocusState::Programmatic);
 
+            // Now that the renderer is set up, update the appearance for initialization
+            UpdateAppearance(_settings);

```

Comment:
okay, so. It looks like we're doing a bunch of other things here that are touching XAML objects. I _think_ that initialization happens **on the UI thread already**. I think that we should then in this case call the "from the UI thread" version. Otherwise, we will end up in a weird place. We're literally saying... "Update the appearance after I return from this function, having unlocked the terminal lock"

I bet that that would deadlock because _UpdateAppearanceFromUIThread takes the lock.

How were the appearance settings getting applied before this? o_O

Reply:
So... I don't remember why I added this line here, and removing it doesn't cause any issues. I'm guessing this line was needed at first but somewhere down the line it wasn't required anymore and I didn't come back to update this! Its gone now :)